pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -224,7 +224,6 @@ private void setUpHelixController() {
     // Emit helix controller metrics
     _controllerMetrics.addCallbackGauge(CommonConstants.Helix.INSTANCE_CONNECTED_METRIC_NAME,
         () -> _helixControllerManager.isConnected() ? 1L : 0L);
-    _controllerMetrics.addCallbackGauge("helix.leader", () -> _helixControllerManager.isLeader() ? 1L : 0L);
```

Comment:
DO not remove the metric, Mark it deprecated.

Reply:
Done.